### PR TITLE
update react-router-redux version 4 

### PR DESCRIPTION
**Pull request by isman-usoh**
- update version 4
- duplicate file for legacy version (2.x - 3.x)

### CODE REVIEW

File: react-router-redux/react-router-redux.d.ts [hunk 1]
```diff
@@ -34,13 +34,13 @@ declare namespace ReactRouterRedux {
...
-    interface HistoryMiddleware extends R.Middleware {
-        listenForReplays(store: R.Store, selectLocationState?: Function): void;
+    interface HistoryMiddleware extends H.History {
         unsubscribe(): void;
     }
 
-    function routeReducer(state?: any, options?: any): R.Reducer;
-    function syncHistory(history: H.History): HistoryMiddleware;
+    function routerReducer(state?: any, options?: any): R.Reducer;
+    function syncHistoryWithStore(history: H.History, store: R.Store, options?: any): HistoryMiddleware;
```
Comment (rosendi) [inline react-router-redux/react-router-redux.d.ts]: With https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8269:
```
interface DefaultSelectLocationState extends Function {
  (state: any): any;
}

interface SyncHistoryWithStoreOptions {
  selectLocationState?: DefaultSelectLocationState;
  adjustUrlOnReplay?: boolean;
}

export var syncHistoryWithStore: (..., options?: SyncHistoryWithStoreOptions) => any; }
```

File: react-router-redux/react-router-redux.d.ts [hunk 2]
```diff
@@ -34,13 +34,13 @@ declare namespace ReactRouterRedux {
         goForward: GoForwardAction;
         goBack: GoBackAction;
     }
-    interface HistoryMiddleware extends R.Middleware {
-        listenForReplays(store: R.Store, selectLocationState?: Function): void;
+    interface HistoryMiddleware extends H.History {
```
Comment (rosendi) [inline react-router-redux/react-router-redux.d.ts]: why not  R.Middleware?

**Comment by rosendi** [comment]
An ideal typings would be a combination with this PR :-) https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8269

File: react-router-redux/react-router-redux-tests.ts [hunk 3]
```diff
@@ -6,15 +6,10 @@
...
-// Sync dispatched route actions to the history
-const reduxRouterMiddleware = syncHistory(browserHistory);
-const createStoreWithMiddleware = applyMiddleware(reduxRouterMiddleware)(createStore);
-
-const store = createStoreWithMiddleware(reducer);
-
-// Required for replaying actions from devtools to
-reduxRouterMiddleware.listenForReplays(store);
+// Create an enhanced history that syncs navigation events with the store
+const history = syncHistoryWithStore(browserHistory, store)
```
Comment (rosendi) [inline react-router-redux/react-router-redux-tests.ts]: I think we have to add some more tests to cover `push(location)`, `replace(location)`, `go(number)`, `goBack()`, `goForward()`, etc... Don't you think so?

File: react-router-redux/react-router-redux.d.ts (hunk 2 above)
Comment (isman-usoh) [inline react-router-redux/react-router-redux.d.ts]: v.4 syncHistoryWithStore return react-router history

Comment (rosendi) [inline react-router-redux/react-router-redux.d.ts]: OK!

**Pull request merged**